src/main/java/org/jenkinsci/plugins/googleplayandroidpublisher/Util.java

```diff
@@ -29,6 +30,8 @@
 import static hudson.Util.fixEmptyAndTrim;
 
 public class Util {
+    private static JenkinsUtil sJenkins = UtilsImpl.getInstance();
```

Comment:
I'm open to other suggestions for these, but I needed to have some way to override the default behavior in unit tests. The implementations have been split into `AndroidUtil` and `JenkinsUtil` interfaces, where `UtilsImpl` contains the default implementations, and `TestUtilImpl` contains the simplified unit test behaviors.

Reply:
Yeah, I'm not really sure what the best way is either.  Jenkins _sort-of_ has dependency injection with Guice, but it's supposedly deprecated (not that it's really mentioned formally anywhere, or Jenkins ever removes deprecated code), and I don't really see any other plugins using it for their tests…

If we had a clear way to do DI, it would be nicer to directly inject the `JenkinsUtil` and `AndroidUtil` into the classes that need them, rather than having this indirection here, which feels a tiny bit odd.

But for now, this seems to work alright!

However, one thing I did notice: I tried modifying the `JenkinsRule` test cases so that the build steps would run on a "remote" machine, and then saw that the default `UtilsImpl` was used again, since a new instance of `Util` was instantiated on the remote machine.